### DISCUSSION TITLE
Data Plane based on OSS proxy

### DISCUSSION THREAD

**Discussion by MoritzKeppler** [eclipse-edc/Connector#1121] [Ideas]
Splitting control and dataplane is one of the central design decisions we took. The DPF (dataplane framework) build on top of the same framework we built for the control plane let us learn what should be part of the data plane (proxy behavior, policy enforcement, token validation etc.).
To be prepared for scenarios with high volume of data / a lot of parallel connections I suggest to evaluate how a data plane that is built around an OSS proxy solution (e.g. nginx, envoy, ... or even a API gw solution?) could look like.
Assume that we need at least some kind of configuration service that translates contracts to proxy configuration and some plugin the intercepts the data flow to evaluate a policy.
Working on an additional implementation of the data plane will also help us to clarify the interface between the control and data plane.

What do you think? Anyone else in favor of exploring this idea?
<sub> Moritz Keppler <moritz.keppler@mercedes-benz.com>, Mercedes-Benz Tech Innovation GmbH, [legal info/Impressum](https://github.com/mercedes-benz/daimler-foss/blob/master/LEGAL_IMPRINT.md)</sub>

**Comment by ndr-brt** [comment]
What about choosing a single OSS proxy solution and spike on it? Probably Nginx is the most used and the most extended.
